### ISSUE TITLE
Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:5858)

### ISSUE THREAD

**Issue by ahmadawais** [Azure/azure-functions-nodejs-worker#89]
I'm facing a weird issue.

Error:

```
Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:5858).
```

```
VSCode 1.23.1
MacOS 
Node: 10.1.0
OS: macOS Sierra
npm: 6.0.1
Chrome: 66.0.3359.181
```

Launch.json

```json
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Attach to JavaScript Functions",
			"type": "node",
			"request": "attach",
			"port": 5858,
			"protocol": "inspector",
			"preLaunchTask": "runFunctionsHost"
		}
	]
}
```

![image](https://on.ahmda.ws/riqo/c)

Trying to create an Azure function.

Several errors in my terminal output:

```sh

> Executing task: func host start <


                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Hosting environment: Production
Content root path: /Users/$USER/Documents/Web/Sandbox/afn
Now listening on: http://localhost:7071
Application started. Press Ctrl+C to shut down.
[5/19/18 8:24:57 PM] Reading host configuration file '/Users/$USER/Documents/Web/Sandbox/afn/host.json'
[5/19/18 8:24:57 PM] Host configuration file read:
[5/19/18 8:24:57 PM] {}
[5/19/18 8:24:57 PM] Starting Host (HostId=ahmadsmacbookpro-763423413, InstanceId=50434726-525b-4afa-9566-a137447a193f, Version=2.0.11651.0, ProcessId=98058, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
[5/19/18 8:24:57 PM] Unable to configure java worker. Could not find JAVA_HOME app setting.
[5/19/18 8:24:57 PM]
[5/19/18 8:24:57 PM] Could not configure language worker Java.
[5/19/18 8:24:57 PM]
[5/19/18 8:24:57 PM] Start Process: node  --inspect=5858 "/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/dist/src/nodejsWorker.js" --host 127.0.0.1 --port 55405 --workerId 33587250-9112-4a07-a373-8572a42396da --requestId 53993c42-bc16-4358-ab64-074a66876e11
[5/19/18 8:24:57 PM] Debugger listening on port 5858.
[5/19/18 8:24:57 PM] Warning: This is an experimental feature and could change at any time.
[5/19/18 8:24:57 PM] To start debugging, open the following URL in Chrome:
[5/19/18 8:24:57 PM]     chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=127.0.0.1:5858/a514a2fa-8b05-47f9-b371-79afd68b44c5
[5/19/18 8:24:57 PM] Generating 1 job function(s)
[5/19/18 8:24:57 PM] Found the following functions:
[5/19/18 8:24:57 PM] Host.Functions.HttpTriggerJS
[5/19/18 8:24:57 PM]
[5/19/18 8:24:57 PM] Host initialized (656ms)
[5/19/18 8:24:57 PM] azure-functions-nodejs-worker officially supports node version >=8.4.0. Current version v7.10.0.
[5/19/18 8:24:57 PM] To install required native modules for v7.10.0, install node-pre-gyp via 'npm i -g node-pre-gyp'.
[5/19/18 8:24:57 PM] Navigate to '<node-worker-dir>/grpc' and run 'node-pre-gyp install'
[5/19/18 8:24:57 PM] Host started (687ms)
[5/19/18 8:24:57 PM] Job host started
[5/19/18 8:24:57 PM] Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/grpc/src/node/extension_binary/node-v51-darwin-x64/grpc_node.node'
[5/19/18 8:24:57 PM] module.js:472
[5/19/18 8:24:57 PM]     throw err;
[5/19/18 8:24:57 PM]     ^
[5/19/18 8:24:57 PM] Error: Cannot find module './Worker.js'
[5/19/18 8:24:57 PM]     at Function.Module._resolveFilename (module.js:470:15)
[5/19/18 8:24:57 PM]     at Function.Module._load (module.js:418:25)
[5/19/18 8:24:57 PM]     at Module.require (module.js:498:17)
[5/19/18 8:24:57 PM]     at require (internal/module.js:20:19)
[5/19/18 8:24:57 PM]     at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/dist/src/nodejsWorker.js:8:14)
[5/19/18 8:24:57 PM]     at Module._compile (module.js:571:32)
[5/19/18 8:24:57 PM]     at Object.Module._extensions..js (module.js:580:10)
[5/19/18 8:24:57 PM]     at Module.load (module.js:488:32)
[5/19/18 8:24:57 PM]     at tryModuleLoad (module.js:447:12)
[5/19/18 8:24:57 PM]     at Function.Module._load (module.js:439:3)
[5/19/18 8:24:57 PM] Worker encountered an error.
[5/19/18 8:24:57 PM] Worker process with pid 98060 exited with code 1.
[5/19/18 8:24:57 PM] Start Process: node  --inspect=5858 "/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/dist/src/nodejsWorker.js" --host 127.0.0.1 --port 55405 --workerId fa3f45e4-eb03-424a-903f-e3e25ee692ad --requestId e3ec9f82-c612-4706-be84-aff784a88ff4
[5/19/18 8:24:57 PM] Debugger listening on port 5858.
[5/19/18 8:24:57 PM] Warning: This is an experimental feature and could change at any time.
[5/19/18 8:24:57 PM] To start debugging, open the following URL in Chrome:
[5/19/18 8:24:57 PM]     chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=127.0.0.1:5858/XXXXX-0e65-47d6-b569-63693636c245
Listening on http://localhost:7071/
Hit CTRL-C to exit...

Http Functions:

        HttpTriggerJS: http://localhost:7071/api/HttpTriggerJS

[5/19/18 8:24:57 PM] azure-functions-nodejs-worker officially supports node version >=8.4.0. Current version v7.10.0.
[5/19/18 8:24:57 PM] To install required native modules for v7.10.0, install node-pre-gyp via 'npm i -g node-pre-gyp'.
[5/19/18 8:24:57 PM] Navigate to '<node-worker-dir>/grpc' and run 'node-pre-gyp install'
[5/19/18 8:24:58 PM] Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/grpc/src/node/extension_binary/node-v51-darwin-x64/grpc_node.node'
[5/19/18 8:24:58 PM] module.js:472
[5/19/18 8:24:58 PM]     throw err;
[5/19/18 8:24:58 PM]     ^
[5/19/18 8:24:58 PM] Error: Cannot find module './Worker.js'
[5/19/18 8:24:58 PM]     at Function.Module._resolveFilename (module.js:470:15)
[5/19/18 8:24:58 PM]     at Function.Module._load (module.js:418:25)
[5/19/18 8:24:58 PM]     at Module.require (module.js:498:17)
[5/19/18 8:24:58 PM]     at require (internal/module.js:20:19)
[5/19/18 8:24:58 PM]     at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/dist/src/nodejsWorker.js:8:14)
[5/19/18 8:24:58 PM]     at Module._compile (module.js:571:32)
[5/19/18 8:24:58 PM]     at Object.Module._extensions..js (module.js:580:10)
[5/19/18 8:24:58 PM]     at Module.load (module.js:488:32)
[5/19/18 8:24:58 PM]     at tryModuleLoad (module.js:447:12)
[5/19/18 8:24:58 PM]     at Function.Module._load (module.js:439:3)
[5/19/18 8:24:58 PM] Worker encountered an error.
[5/19/18 8:24:58 PM] Worker process with pid 98062 exited with code 1.
[5/19/18 8:24:58 PM] Start Process: node  --inspect=5858 "/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/dist/src/nodejsWorker.js" --host 127.0.0.1 --port 55405 --workerId e50dd0da-8419-49b8-8a25-87520752a694 --requestId 71f7ab6c-a836-4d57-b0f8-8f9d541f6b4d
[5/19/18 8:24:58 PM] Debugger listening on port 5858.
[5/19/18 8:24:58 PM] Warning: This is an experimental feature and could change at any time.
[5/19/18 8:24:58 PM] To start debugging, open the following URL in Chrome:
[5/19/18 8:24:58 PM]     chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=127.0.0.1:5858/as-631d-4dfe-b7da-7e9ccf4379e9
[5/19/18 8:24:58 PM] azure-functions-nodejs-worker officially supports node version >=8.4.0. Current version v7.10.0.
[5/19/18 8:24:58 PM] To install required native modules for v7.10.0, install node-pre-gyp via 'npm i -g node-pre-gyp'.
[5/19/18 8:24:58 PM] Navigate to '<node-worker-dir>/grpc' and run 'node-pre-gyp install'
[5/19/18 8:24:58 PM] Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/grpc/src/node/extension_binary/node-v51-darwin-x64/grpc_node.node'
[5/19/18 8:24:58 PM] module.js:472
[5/19/18 8:24:58 PM]     throw err;
[5/19/18 8:24:58 PM]     ^
[5/19/18 8:24:58 PM] Error: Cannot find module './Worker.js'
[5/19/18 8:24:58 PM]     at Function.Module._resolveFilename (module.js:470:15)
[5/19/18 8:24:58 PM]     at Function.Module._load (module.js:418:25)
[5/19/18 8:24:58 PM]     at Module.require (module.js:498:17)
[5/19/18 8:24:58 PM]     at require (internal/module.js:20:19)
[5/19/18 8:24:58 PM]     at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.25/workers/node/dist/src/nodejsWorker.js:8:14)
[5/19/18 8:24:58 PM]     at Module._compile (module.js:571:32)
[5/19/18 8:24:58 PM]     at Object.Module._extensions..js (module.js:580:10)
[5/19/18 8:24:58 PM]     at Module.load (module.js:488:32)
[5/19/18 8:24:58 PM]     at tryModuleLoad (module.js:447:12)
[5/19/18 8:24:58 PM]     at Function.Module._load (module.js:439:3)
[5/19/18 8:24:58 PM] Worker encountered an error.
[5/19/18 8:24:58 PM] Worker process with pid 98063 exited with code 1.

```

Help.

Part of it was fixed here https://github.com/Microsoft/vscode-azurefunctions/issues/387#issuecomment-393973376 and I was asked to move the issue to this repo.

**Comment by mhoeger** [comment]
@ahmadawais - If you switch to node 8.x, you should be fine. I'm sorry about the confusion :( we've recently updated the error message you saw to now say this:
https://github.com/Azure/azure-functions-nodejs-worker/blob/0d5d754087baa524e9d636a4ec89b5e08a314df3/src/GrpcService.ts#L3-L5

And we're also working on supporting node 10 shortly!

**Comment by mhoeger** [comment]
Closing this issue for now, but please re-open and ping me if you are still having issues!

**Comment by ahmadawais** [comment]
@mhoeger sadly I tried to fix it. And I'm still stuck.

My node version is `v8.11.2`
Extension Azure Functions `0.9.1`
azure-functions-core-tools `2.0.1-beta.28`

I am on MacOS.

Here's the output now.

```sh
> Executing task: func host start <


                  %%%%%%
                 %%%%%%
            @   %%%%%%    @
          @@   %%%%%%      @@
       @@@    %%%%%%%%%%%    @@@
     @@      %%%%%%%%%%        @@
       @@         %%%%       @@
         @@      %%%       @@
           @@    %%      @@
                %%
                %

Hosting environment: Production
Content root path: /Users/ahmadawais/Documents/Web/Sandbox/demoazurefunc
Now listening on: http://localhost:7071
Application started. Press Ctrl+C to shut down.
[6/8/18 10:00:23 PM] Reading host configuration file '/Users/ahmadawais/Documents/Web/Sandbox/demoazurefunc/host.json'
[6/8/18 10:00:23 PM] Host configuration file read:
[6/8/18 10:00:23 PM] {}
[6/8/18 10:00:23 PM] Starting Host (HostId=ahmadsmacbookpro-552755506, InstanceId=57b3f52e-2e98-4014-b96c-a7b0b191533f, Version=2.0.11776.0, ProcessId=67768, AppDomainId=1, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
[6/8/18 10:00:24 PM] Loading all the worker providers from the default workers directory: /usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers
[6/8/18 10:00:24 PM] Found worker config: /usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/python/worker.config.json
[6/8/18 10:00:24 PM] Will load worker provider for language: python
[6/8/18 10:00:24 PM] Start Process: node  --inspect=5858 "/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js" --host 127.0.0.1 --port 59035 --workerId 2dfd6191-c937-434b-b534-00aeec4ac846 --requestId 3249c043-f995-43ba-bf3e-5e6d0750ecce --grpcMaxMessageLength 134217728
[6/8/18 10:00:24 PM] Debugger listening on ws://127.0.0.1:5858/38b6875c-cf13-4828-9709-a4bd3ae00629
[6/8/18 10:00:24 PM] For help, see: https://nodejs.org/en/docs/inspector
[6/8/18 10:00:24 PM] Generating 1 job function(s)
[6/8/18 10:00:24 PM] Found the following functions:
[6/8/18 10:00:24 PM] Host.Functions.HttpTriggerJS
[6/8/18 10:00:24 PM]
[6/8/18 10:00:24 PM] Host initialized (829ms)
[6/8/18 10:00:24 PM] Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/grpc/src/node/extension_binary/node-v64-darwin-x64/grpc_node.node'
[6/8/18 10:00:24 PM] internal/modules/cjs/loader.js:573
[6/8/18 10:00:24 PM]     throw err;
[6/8/18 10:00:24 PM]     ^
[6/8/18 10:00:24 PM] Error: Cannot find module './Worker.js'
[6/8/18 10:00:24 PM]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:571:15)
[6/8/18 10:00:24 PM]     at Function.Module._load (internal/modules/cjs/loader.js:497:25)
[6/8/18 10:00:24 PM]     at Module.require (internal/modules/cjs/loader.js:626:17)
[6/8/18 10:00:24 PM]     at require (internal/modules/cjs/helpers.js:20:18)
[6/8/18 10:00:24 PM]     at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js:8:14)
[6/8/18 10:00:24 PM]     at Module._compile (internal/modules/cjs/loader.js:678:30)
[6/8/18 10:00:24 PM]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
[6/8/18 10:00:24 PM]     at Module.load (internal/modules/cjs/loader.js:589:32)
[6/8/18 10:00:24 PM]     at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
[6/8/18 10:00:24 PM]     at Function.Module._load (internal/modules/cjs/loader.js:520:3)
[6/8/18 10:00:24 PM] Host started (880ms)
[6/8/18 10:00:24 PM] Job host started
Listening on http://localhost:7071/
Hit CTRL-C to exit...

Http Functions:

        HttpTriggerJS: http://localhost:7071/api/HttpTriggerJS

[6/8/18 10:00:24 PM] Worker encountered an error.
[6/8/18 10:00:24 PM] Worker process with pid 67778 exited with code 1.
[6/8/18 10:00:24 PM] Start Process: node  --inspect=5858 "/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js" --host 127.0.0.1 --port 59035 --workerId 2a29965b-a6a3-44c5-8299-338ae5f11b6c --requestId f12448a6-ed27-4d58-bfad-e7388aa55cc6 --grpcMaxMessageLength 134217728
[6/8/18 10:00:24 PM] Debugger listening on ws://127.0.0.1:5858/1ac024e1-2b4c-4c68-b967-bcad8b886729
[6/8/18 10:00:24 PM] For help, see: https://nodejs.org/en/docs/inspector
[6/8/18 10:00:24 PM] Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/grpc/src/node/extension_binary/node-v64-darwin-x64/grpc_node.node'
[6/8/18 10:00:24 PM] internal/modules/cjs/loader.js:573
[6/8/18 10:00:24 PM]     throw err;
[6/8/18 10:00:24 PM]     ^
[6/8/18 10:00:24 PM] Error: Cannot find module './Worker.js'
[6/8/18 10:00:24 PM]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:571:15)
[6/8/18 10:00:24 PM]     at Function.Module._load (internal/modules/cjs/loader.js:497:25)
[6/8/18 10:00:24 PM]     at Module.require (internal/modules/cjs/loader.js:626:17)
[6/8/18 10:00:24 PM]     at require (internal/modules/cjs/helpers.js:20:18)
[6/8/18 10:00:24 PM]     at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js:8:14)
[6/8/18 10:00:24 PM]     at Module._compile (internal/modules/cjs/loader.js:678:30)
[6/8/18 10:00:24 PM]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
[6/8/18 10:00:24 PM]     at Module.load (internal/modules/cjs/loader.js:589:32)
[6/8/18 10:00:24 PM]     at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
[6/8/18 10:00:24 PM]     at Function.Module._load (internal/modules/cjs/loader.js:520:3)
[6/8/18 10:00:24 PM] Worker encountered an error.
[6/8/18 10:00:24 PM] Worker process with pid 67781 exited with code 1.
[6/8/18 10:00:24 PM] Start Process: node  --inspect=5858 "/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js" --host 127.0.0.1 --port 59035 --workerId 459cf8d4-6780-4097-b860-5acc0d5f2784 --requestId c4d945b8-c07a-4f09-a86c-5f803f594803 --grpcMaxMessageLength 134217728
[6/8/18 10:00:24 PM] Debugger listening on ws://127.0.0.1:5858/66a5bdb5-aff1-456e-bd13-f2e487008cc4
[6/8/18 10:00:24 PM] For help, see: https://nodejs.org/en/docs/inspector
[6/8/18 10:00:25 PM] Debugger attached.
[6/8/18 10:00:25 PM] Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/grpc/src/node/extension_binary/node-v64-darwin-x64/grpc_node.node'
[6/8/18 10:00:25 PM] internal/modules/cjs/loader.js:573
[6/8/18 10:00:25 PM]     throw err;
[6/8/18 10:00:25 PM]     ^
[6/8/18 10:00:25 PM] Error: Cannot find module './Worker.js'
[6/8/18 10:00:25 PM]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:571:15)
[6/8/18 10:00:25 PM]     at Function.Module._load (internal/modules/cjs/loader.js:497:25)
[6/8/18 10:00:25 PM]     at Module.require (internal/modules/cjs/loader.js:626:17)
[6/8/18 10:00:25 PM]     at require (internal/modules/cjs/helpers.js:20:18)
[6/8/18 10:00:25 PM]     at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js:8:14)
[6/8/18 10:00:25 PM]     at Module._compile (internal/modules/cjs/loader.js:678:30)
[6/8/18 10:00:25 PM]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
[6/8/18 10:00:25 PM]     at Module.load (internal/modules/cjs/loader.js:589:32)
[6/8/18 10:00:25 PM]     at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
[6/8/18 10:00:25 PM]     at Function.Module._load (internal/modules/cjs/loader.js:520:3)
[6/8/18 10:00:25 PM] Waiting for the debugger to disconnect...
[6/8/18 10:00:25 PM] Worker encountered an error.
[6/8/18 10:00:25 PM] Worker process with pid 67782 exited with code 1.
```

and this is what I get in debug console…

```js
Couldn't require bundle, falling back to Worker.js. Error: Cannot find module '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/grpc/src/node/extension_binary/node-v64-darwin-x64/grpc_node.node'
nodejsWorker.js:7
Error: Cannot find module './Worker.js'
loader.js:571
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:571:15)
    at Function.Module._load (internal/modules/cjs/loader.js:497:25)
    at Module.require (internal/modules/cjs/loader.js:626:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.28/workers/node/dist/src/nodejsWorker.js:8:14)
    at Module._compile (internal/modules/cjs/loader.js:678:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:689:10)
    at Module.load (internal/modules/cjs/loader.js:589:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:528:12)
    at Function.Module._load (internal/modules/cjs/loader.js:520:3)

```

**Comment by ahmadawais** [comment]
@mhoeger yes it is strange.

I tried everything again. No luck.

I made a video of the process. Using VSCode here not the `func` CLI.

VIDEO: https://on.ahmda.ws/s98K

**Comment by ahmadawais** [comment]
Oh wow! I just figured it out. It's working fine now.

So, here's what happened. I have `node -v` reporting version `8.11.2` everywhere in the VSCode terminal, and in iTerm2. But I had another project open in VSCode which I was working on earlier. That had the terminal open as well. So, when I tried, that one reported `10.1.0` so instead of reloading VSCode, I quit and started it from scratch. That fixed it.
